llvm/lib/Transforms/Instrumentation/ThreadSanitizer.cpp

```diff
@@ -84,6 +88,15 @@ static cl::opt<bool>
     ClOmitNonCaptured("tsan-omit-by-pointer-capturing", cl::init(true),
                       cl::desc("Omit accesses due to pointer capturing"),
                       cl::Hidden);
+static cl::opt<bool>
```

Comment:
It would be useful to have a meaningful description of what exactly this is doing somewhere in the source code (probably here). The description may be condensed from the PR description.
Just repeating the variable/flag name (dominance analysis) won't tell me much in future.
It also needs to say why this is disabled by default, and what's the plan for enabling it in future (are there any major unresolved problems, or it's just not enabled to do more gradual enabling). Note that if it's disabled by default, it won't really get lots of testing out there, and won't bring much benefit (almost nobody will enable this).
Also it needs to detail what's exactly unsafe in tsan-postdom-aggressive.